### ISSUE TITLE
UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`.

### ISSUE THREAD

**Issue by huuquan1994** [xinntao/BasicSR#206]
Hi @xinntao ,

I ran your code using Pytorch 1.2.0 and got the warning 

> UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`.

This warning is because of the [train.py](https://github.com/xinntao/BasicSR/blob/60d1bc0c7b0b787783b164b11add23a70cc5e565/codes/train.py#L156) call the update learning rate before training (please see the line 156). ==> `lr_scheduler` in the `update_learning_rate` function of the [base_model.py](https://github.com/xinntao/BasicSR/blob/0e7d000bece795ea68d8de138bf574a56de9a081/codes/models/base_model.py#L51) is called before the `optimizer` is called.

When I moved the "update learning rate" (line 157) to be below the "training code" (line 160 & 161), the warning disappeared. 
I wonder if that action affects the training result?

**Comment by xinntao** [comment]
It will not affect the training result. 
We put it before updating the optimizer because we implement the warm-up in the `update_learning_rate`.
